src/libsystemd/sd-netlink/nfnl-message.c

```diff
@@ -96,15 +97,13 @@ int sd_nfnl_nft_message_new_basechain(sd_netlink *nfnl, sd_netlink_message **ret
                                       const char *table, const char *chain,
                                       const char *type,
                                       uint8_t hook, int prio) {
-        sd_netlink_message *m;
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *m = NULL;
         int r;
 
-        r = nft_message_new(nfnl, ret, family, NFT_MSG_NEWCHAIN, NLM_F_CREATE | NLM_F_ACK);
+        r = nft_message_new(nfnl, &m, family, NFT_MSG_NEWCHAIN, NLM_F_CREATE | NLM_F_ACK);
         if (r < 0)
                 return r;
 
-        m = *ret;
-
```

Comment:
this looks like a broken squahs: should have been squashed into the earlier commit that added the function?

Reply:
Yes, exactly. I've fixed the squash so that all of the added code has the same pattern from the start, i.e.:
_cleanup_(sd_netlink_message_unrefp) sd_netlink_message *m = NULL;
...
*ret = TAKE_PTR(m);
return 0.